### ISSUE TITLE
Where to keep the elementary constants?

### ISSUE THREAD

**Issue by keilw** [unitsofmeasurement/indriya#43]
The SI redefinition https://www.euramet.org/publications-media-centre/news/?tx_news_pi1%5Bnews%5D=637&tx_news_pi1%5Bcontroller%5D=News&tx_news_pi1%5Baction%5D=detail&cHash=d1b21b8e8a4522f5cfcf3b101a095f04 makes elementary constants like the **Avogadro constant**, **Planck constant** or **Elementary charge** more important than they already used to be.

Right now a few of them are defined internally in the [NonSI](https://github.com/unitsofmeasurement/si-units/blob/master/units/src/main/java/si/uom/NonSI.java) class of the SI Units module. 

Given their increased role, we could either:

**Comment by keilw** [comment]
A) Keep them in the SI Units module, but probably move to the `SI` class with public visibility. 
     Here it should be possible to define most of them as units rather than numerical constants.

**Comment by keilw** [comment]
B) Define them in the RI (here) e.g. in `AbstractSystemOfUnits` or `Units`.
    Given they involve and describe the **SI base Units**, it should be possible to define them as units, here, too.

**Comment by keilw** [comment]
C) Define them in the API, e.g. in the `SystemOfUnits` interface. There we may only be able to define their numerical value similar to
```
 /**
     * Holds the Avogadro constant.
     */
    private static final double AVOGADRO_CONSTANT = 6.02214199e23; // (1/mol).
```
from the `NonSI` class of the current SI Units module, but `public`.

**Comment by filipvanlaenen** [comment]
My rationale for voting C down: seems wrong to define them as numerical values only in an API. Isn't this an interface that can be implemented by systems of units in which the numerical values wouldn't make sense because of conversions?

Option A seems better than B. But since someone decided earlier to go for option A for the non-SI constants, and it has worked fine so far, I tend to go for the option that simply extends that decision. But si-units is supposed to be “A library of SI quantities and unit types”, and to me, the definition of constants doesn't seem to be something that has something to do with a reference implementation either. (Actually, I feel even stronger for A after writing this down.)

**Comment by keilw** [comment]
Good point, thanks. I have not voted myself, but will see, how others do. While Indriya covers all relevant SI base units and quantities, I also lean towards using the SI module.

**Comment by dautelle** [comment]
My preference goes to A, with the following caveat: Only constants with fixed numerical values defined by SI should go in SI.
